### DISCUSSION TITLE
Updating Streetmerchant

### DISCUSSION THREAD

**Discussion by mikeike951** [jef/streetmerchant#1416] [General]
I apologize if this sounds completely like a noob, as I'm definitely a noob when it comes to command prompt, but I've seen updates to streetmerchant to resolve some of the issues that I'm having, but no way to update it. Do we need to do anything on our end to update it, or is that handled before it gets to me?

**Comment by airsharkie** [comment]
Hi, run this from the command line, from:https://github.com/jef/streetmerchant/wiki/Installation:-Windows

Bot Update
End the bot by pressing Ctrl + C.
Type git pull origin main
Type npm install (only required if there are new dependencies)
Type npm run start

**Reply by mikeike951** [reply]
I've tried doing this, but this is what pulls up after the pull command.
![image](https://user-images.githubusercontent.com/76020129/102297760-0e2eb980-3f0d-11eb-8dd2-95df8d62775a.png)

**Comment by CorgFather** [comment]
@mikeike951 

Are the changes you mention in versions higher than 3.2.0? There's a file version.txt in the streetmerchant install folder. If so, the command might be a bit different. I'd defer to @airsharkie on how, as this was my first time using git command line and am on Mac.

**Reply by mikeike951** [reply]
I am meaning higher than 3.2

**Reply by airsharkie** [reply]
I'm no pro at Git, but I ended up Googling the issue after I found the same thing.

It's telling us that it doesn't want to overwrite a file because it's been personalized by us somehow.
I'm sure there are 100 ways to do this, "stashing" the custom file, then restoring it later, but here's the command I went with to make it work:

(My "warning" file was 'package-lock.json')

1. CTRL+C to quit Streetmerchant if it's running
2. git checkout HEAD^ package-lock.json
3. git pull origin main
4. npm install (only required if there are new dependencies)
5. npm run start

Mine upgraded to version 3.30 with no problem after this.

Let me know if this works for you.

**Reply by mikeike951** [reply]
That did work! Thank you so much! I definitely need to learn how to work with command line a lot more.
